### PR TITLE
Include install_requires and replace requirements.txt

### PR DESCRIPTION
**Pull request by jtratner**
So it stays a well-behaved python package

1. Stop using no-deps for main install
2. move requirements.txt into install_requires

### CODE REVIEW

**Comment by jtratner** [comment]
hmm - been a lot of drift since last time tests passed (last year) - going to hand this off to other folks like @hrichards . We shouldn't get rid of install_requires - definitely necessary to be able to use baya effectively.

**Comment by hrichards** [comment]
Hey @jtratner, very nice work on getting the setup in Baya modernized!

I took what you started with and ran with it to try to get a green build for this library.  By moving over my changes from https://github.com/counsyl/baya/pull/20, updating a call to `django-auth-ldap:authenticate` in the Baya tests in a way justified in the `django-auth-ldap` release notes, and dropping unsupported Django versions, I think I've got a justifiable green build here.

Your changes/commits LGTM; what do you think of mine?

**Review by krhaas — COMMENTED** [review]
Nice. changes look good to me too.

**Review by jdavisp3 — APPROVED** [review]
🚀 Thanks @hrichards and @jtratner !

**Comment by hrichards** [comment]
Thanks, @jdavisp3 and @krhaas!  I'm going to change the version declared in these changes to 1.2.0 since we're dropping support for Django versions and that's backwards-incompatible, but after I push that I'll merge.

Really appreciate the help here.

File: CHANGELOG [hunk 1]
```diff
@@ -1,4 +1,4 @@
-1.1.2: Change `initialize` declaration to be compatible with django-auth-ldap>=1.5.0.
+1.2.0: Drop unsupported Django versions.  Redo setup system.  Change `initialize` and `authenticate` declaration to be compatible with django-auth-ldap>=1.5.0.
```
Comment (lucaswiman) [inline CHANGELOG:1]: This should be 2.X in semver. (Breaking.Feature.Fix)

**Pull request merged**